src/AWSMetadataUtilities.jl

```diff
@@ -0,0 +1,578 @@
+module AWSMetadataUtilities
+
+include("AWSExceptions.jl")
+
+using .AWSExceptions
+using OrderedCollections: OrderedDict, LittleDict
+using HTTP
+using JSON
+using Mocking
+
+
+"""
+    _get_aws_sdk_js_files()
+
+Get a list of all AWS service API definition files from the `awsk-sdk-js` GitHub repository.
+
+# Returns
+- `Array{OrderedDict}`: Array of Dictionaries which contains information about each AWS Service
+"""
+function _get_aws_sdk_js_files()
+    headers = ["User-Agent" => "JuliaCloud/AWS.jl"]
+    url = "https://api.github.com/repos/aws/aws-sdk-js/contents/apis"
+
+    req = @mock HTTP.get(url, headers)
+    files = JSON.parse(String(req.body), dicttype=OrderedDict)
+    filter!(f -> endswith(f["name"], ".normal.json"), files)  # Only get ${Service}.normal.json files
+    files = _filter_latest_service_version(files)
+
+    return files
+end
+
+"""
+    _filter_latest_service_version(services::Array{Any})
+
+Return a list of all AWS Services and their latest version.
+
+# Arguments
+- `services::Array`: Array of all AWS APIs and their versions
+
+# Returns
+- `Array{Dict}`: Array of the latest AWS Service definitions
+"""
+function _filter_latest_service_version(services::Array)
+    seen_services = String[]
+    latest_versions = OrderedDict[]
+
+    services = reverse(services)
+
+    for service in services
+        service_name, _ = _get_service_and_version(service["name"])
+
+        if !(service_name in seen_services)
+            push!(seen_services, service_name)
+            push!(latest_versions, service)
+        end
+    end
+
+    return latest_versions
+end
+
+"""
+    _get_service_and_version(filename::String)
+
+Get the `service` and `version` from a filename.
+
+# Arguments
+- `filename`: Name of the file, ex. `{Service}-{Version}.normal.json`
+
+# Returns
+- `Tuple{String, String}`: `service` and `version` for an AWS service
+"""
+function _get_service_and_version(filename::String)
+    try
+        service_and_version = join(split(filename, '.')[1:end-2],'.')
+        service_and_version = split(service_and_version, '-')
+        service = join(service_and_version[1:end-3], '-')
+        version = join(service_and_version[end-2:end], '-')
```

Comment:
State the intent to be clear what we want to do
```suggestion
        # Remove ".normal.json" suffix
        service_and_version = join(split(filename, '.')[1:end-2],'.')
        
        service_and_version = split(service_and_version, '-')
        service = join(service_and_version[1:end-3], '-')
        version = join(service_and_version[end-2:end], '-')
```